### ISSUE TITLE
Dashboard showing socker error, 

### ISSUE THREAD

**Issue by chaithrar** [riemann/riemann-dash#105]
Hi 

We are using riemann for first time and after setting up, we are seeing socket error check server field.
Trying to add the events via ruby client but still not successful. Riemann is of latest version

Below id the riemann.config file being used Please help us in understanding what is wrong with the configuration.

; -*- mode: clojure; -*-
; vim: filetype=clojure

(logging/init {:file "riemann.log"})

; Listen on the local interface over TCP (5555), UDP (5555), and websockets
; (5556)
(let [host "0.0.0.0"]
  (tcp-server {:host host})
  (udp-server {:host host})
  (ws-server  {:host host}))

; Expire old events from the index every 5 seconds.
(periodically-expire 5)

(let [index (default :ttl 300 (update-index (index)))]
    (streams
      index))

(let [hosts (atom #{})]
  (fn [event]
    (swap! hosts conj (:host event))
    (prn :hosts @hosts)
    (index {:service "unique hosts"
            :time (unix-time)
            :metric (count @hosts)})))

**Comment by aphyr** [comment]
On 12/14/2015 01:43 PM, chaithrar wrote:
> Hi
>
> We are using riemann for first time and after setting up, we are seeing socket
> error check server field
> Trying to add the events via ruby client but still not successful
>
> Below id the riemannconfig file being used Please help us in understanding what
> is wrong with the configuration
>
> ; -/- mode: clojure; -/-
> ; vim: filetype=clojure
>
> (logging/init {:file "riemannlog"})
>
> ; Listen on the local interface over TCP (5555), UDP (5555), and websockets
> ; (5556)
> (let host "0000"

Is "0000" actually your hostname?

--Kyle

**Comment by chaithrar** [comment]
Its not I changed that to actual host ip. But still I am getting the same error from UI

**Comment by chaithrar** [comment]
; -*- mode: clojure; -*-
; vim: filetype=clojure

(logging/init {:file "riemann.log"})

; Listen on the local interface over TCP (5555), UDP (5555), and websockets
; (5556)
(let [host "10.12.20.1"]
  (tcp-server {:host host})
  (udp-server {:host host})
  (ws-server  {:host host}))

; Expire old events from the index every 5 seconds.
(periodically-expire 5)

(let [index (default :ttl 300 (update-index (index)))]
    (streams
      index))

(let [hosts (atom #{})]
  (fn [event]
    (swap! hosts conj (:host event))
    (prn :hosts @hosts)
    (index {:service "unique hosts"
            :time (unix-time)
            :metric (count @hosts)})))

**Comment by aphyr** [comment]
It also looks like you're missing all the brackets from your let bindings? Is your Riemann host reachable by your browser?
On Dec 15, 2015 08:00, chaithrar <notifications@github.com> wrote:; -- mode: clojure; --
; vim: filetype=clojure

(logging/init {:file "riemann.log"})

; Listen on the local interface over TCP (5555), UDP (5555), and websockets
; (5556)
(let host "10.12.20.1"
  (udp-server {:host host})
  (ws-server  {:host host}))

; Expire old events from the index every 5 seconds.
(periodically-expire 5)

(let index (default :ttl 300 (update-index (index))))

(let hosts (atom #{})
            :metric (count @hosts)})))

**Comment by chaithrar** [comment]
WebSocket connection to 'ws://127.0.0.1:5556/index?subscribe=true&query=true' failed: Error in connection establishment: net::ERR_CONNECTION_REFUSED


This is the errror I am getting the browser is trying to make connection to my machine rather than ther server how to configure dashboard to connect to specific server?

**Comment by aphyr** [comment]
Like the error message says: look at the server field in the top right side of the dash.

**Comment by Jovons** [comment]
Today I met the same problem. On the host I could not visit the VM riemann dashboard. After I check VM `netstat -ano | grep 4567`, it shows listening on 0.0.0.0 and port 4567, so I changed the firewall policy. Then I opened browser in host and visit dashboard through http://IP:4567, it shows the same error. Then I noticed the ip address on the top right is 127.0.0.1, and I see network connection through the browser, it was trying to connect to 127.0.0.1 websocket, so I changed that IP address on the top right to the VM IP address. Then it works now.
